python/cudf/cudf/_lib/pylibcudf/libcudf/round.pxd

```diff
@@ -9,9 +9,9 @@ from cudf._lib.pylibcudf.libcudf.column.column_view cimport column_view
 
 cdef extern from "cudf/round.hpp" namespace "cudf" nogil:
 
-    ctypedef enum rounding_method "cudf::rounding_method":
-        HALF_UP "cudf::rounding_method::HALF_UP"
```

Comment:
Do the strings after the enum values mean anything (e.g. "cudf::rounding_method::HALF_UP") or are they purely cosmetic?

(asking since I don't want to unintentionally change behavior by changing this here)

Reply:
The strings are meaningful, but removing them is the right answer. [The strings are a way to alias the names of Cython objects so that they appear differently in generated C code](https://cython.readthedocs.io/en/latest/src/userguide/external_C_code.html#resolving-naming-conflicts-c-name-specifications). IOW when you write `cdef rounding_method x = HALF_UP` in Cython code, that will translate to C code that says `enum rounding_method X = cudf::rounding_method::HALF_UP`. The reason that we previously need this in enums in our Cython is because prior to version 3.0 Cython did not support C++ scoped enumerations (`enum class`), but all of our C++ enums are defined that way, so the alias was the only way to trick Cython into generating the code that we needed (with raw C enums you don't need to do `cudf::rounding_method::HALF_UP`, you can just do `cudf::HALF_UP` because the enum doesn't create it's own namespace). By switching to using an `enum class` here we remove the need for that.